### PR TITLE
dxns bond module test

### PR DESCRIPTION
**Pull request by gsk967**
Closes: #https://github.com/vulcanize/dxns/issues/5

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by gsk967** [comment]
@anilCSE @ashwinphatak  can you review this pull request

File: proto/vulcanize/bond/v1beta1/query.proto [hunk 1]
```diff
@@ -13,27 +13,27 @@ option go_package = "github.com/tharsis/ethermint/x/bond/types";
...
   // GetBondById
   rpc GetBondById(QueryGetBondByIdRequest) returns (QueryGetBondByIdResponse){
-    option (google.api.http).get = "/ethermint/bond/v1/bonds/{id}";
+    option (google.api.http).get = "/vulcanize/bond/v1beta1/bonds/{id}";
   }
 
   // Get Bonds List by Owner
   rpc GetBondsByOwner(QueryGetBondsByOwnerRequest) returns (QueryGetBondsByOwnerResponse){
-    option (google.api.http).get = "/ethermint/bond/v1/bonds_by_owner";
+    option (google.api.http).get = "/vulcanize/bond/v1beta1/bonds_by_owner";
```
Comment (anilCSE) [inline proto/vulcanize/bond/v1beta1/query.proto:31]: This url should have ownder-id in the params

File: x/bond/client/testutil/grpc_gateway_test.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+			"invalid request with headers",
+			fmt.Sprintf("%s/vulcanize/bond/v1beta1/bonds_by_owner", val.APIAddress),
+			map[string]string{
+				grpctypes.GRPCBlockHeightHeader: "1",
+			},
+			0,
+			true,
+		},
+		{
+			"valid request",
```
Comment (anilCSE) [inline x/bond/client/testutil/grpc_gateway_test.go:112]: Only optional params should be considered for query params. Change this to route param instead

**Review by anilCSE — CHANGES_REQUESTED** [review]

File: proto/vulcanize/bond/v1beta1/query.proto (hunk 1 above)
Comment (gsk967) [reply proto/vulcanize/bond/v1beta1/query.proto:31]: `Owner-ID` will be come from Query Params for this route  
`/vulcanize/bond/v1beta1/bonds_by_owner?owner={OwnerAddress}`

File: x/bond/client/testutil/grpc_gateway_test.go (hunk 2 above)
Comment (gsk967) [reply x/bond/client/testutil/grpc_gateway_test.go:112]: Okay

**Comment by gsk967** [comment]
# Testing Commands 

## Test 
```bash
$ go test -v github.com/tharsis/ethermint/x/bond/client/testutil
=== RUN   TestIntegrationTestSuite
    query_test.go:37: setting up integration test suite
    network.go:184: acquiring test network lock
    network.go:189: created temporary directory: /tmp/TestIntegrationTestSuite3172147603/001/ethermint_9000-3111107155468
    network.go:198: preparing test network...
    network.go:246: tcp://0.0.0.0:43615
    network.go:395: starting test network...
    network.go:400: started test network
=== RUN   TestIntegrationTestSuite/TestGetBondByIdGRPC
=== RUN   TestIntegrationTestSuite/TestGetBondByIdGRPC/invalid_request_with_headers
=== RUN   TestIntegrationTestSuite/TestGetBondByIdGRPC/valid_request
=== RUN   TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC
=== RUN   TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC/invalid_request_with_headers
=== RUN   TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC/valid_request
=== RUN   TestIntegrationTestSuite/TestGetBondsByOwnerGRPC
=== RUN   TestIntegrationTestSuite/TestGetBondsByOwnerGRPC/invalid_request_with_headers
=== RUN   TestIntegrationTestSuite/TestGetBondsByOwnerGRPC/valid_request
=== RUN   TestIntegrationTestSuite/TestGetBondsGRPC
=== RUN   TestIntegrationTestSuite/TestGetBondsGRPC/invalid_request_with_headers
=== RUN   TestIntegrationTestSuite/TestGetBondsGRPC/valid_request
=== RUN   TestIntegrationTestSuite/TestGetCmdQueryParams
=== RUN   TestIntegrationTestSuite/TestGetCmdQueryParams/json_output
=== RUN   TestIntegrationTestSuite/TestGetParamsGRPC
=== RUN   TestIntegrationTestSuite/TestGetParamsGRPC/valid_request
=== RUN   TestIntegrationTestSuite/TestGetQueryBondById
=== RUN   TestIntegrationTestSuite/TestGetQueryBondById/Case_invalid_bond_id
=== RUN   TestIntegrationTestSuite/TestGetQueryBondById/Case_create_and_get_bond_by_id
=== RUN   TestIntegrationTestSuite/TestGetQueryBondLists
=== RUN   TestIntegrationTestSuite/TestGetQueryBondLists/Case_create_and_get_bond_lists
=== RUN   TestIntegrationTestSuite/TestGetQueryBondListsByOwner
=== RUN   TestIntegrationTestSuite/TestGetQueryBondListsByOwner/Case_invalid_owner_address
=== RUN   TestIntegrationTestSuite/TestGetQueryBondListsByOwner/Case_get_bond_lists_by_owner_address
=== RUN   TestIntegrationTestSuite/TestGetQueryBondModuleBalance
=== RUN   TestIntegrationTestSuite/TestGetQueryBondModuleBalance/Case_get_bond_module_balance
=== RUN   TestIntegrationTestSuite/TestTxCancelBond
=== RUN   TestIntegrationTestSuite/TestTxCancelBond/Case_without_bond_id
=== RUN   TestIntegrationTestSuite/TestTxCancelBond/Case_cancel_bond
=== RUN   TestIntegrationTestSuite/TestTxCreateBond
=== RUN   TestIntegrationTestSuite/TestTxCreateBond/Case_without_deposit
=== RUN   TestIntegrationTestSuite/TestTxCreateBond/Case_create_bond
=== RUN   TestIntegrationTestSuite/TestTxRefillBond
=== RUN   TestIntegrationTestSuite/TestTxRefillBond/Case_without_refill_amount_and_bond_id
=== RUN   TestIntegrationTestSuite/TestTxRefillBond/Case_refill_bond
=== RUN   TestIntegrationTestSuite/TestTxWithdrawAmountFromBond
=== RUN   TestIntegrationTestSuite/TestTxWithdrawAmountFromBond/Case_without_withdraw_amount_and_bond_id
=== RUN   TestIntegrationTestSuite/TestTxWithdrawAmountFromBond/Case_withdraw_amount_from_bond
=== CONT  TestIntegrationTestSuite
    query_test.go:49: tearing down integration test suite
    network.go:487: cleaning up test network...
    network.go:511: finished cleaning up test network
    network.go:484: released test network lock
--- PASS: TestIntegrationTestSuite (37.57s)
    --- PASS: TestIntegrationTestSuite/TestGetBondByIdGRPC (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondByIdGRPC/invalid_request_with_headers (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondByIdGRPC/valid_request (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC/invalid_request_with_headers (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondModuleBalanceGRPC/valid_request (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetBondsByOwnerGRPC (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondsByOwnerGRPC/invalid_request_with_headers (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondsByOwnerGRPC/valid_request (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetBondsGRPC (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondsGRPC/invalid_request_with_headers (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetBondsGRPC/valid_request (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetCmdQueryParams (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetCmdQueryParams/json_output (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetParamsGRPC (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetParamsGRPC/valid_request (0.00s)
    --- PASS: TestIntegrationTestSuite/TestGetQueryBondById (5.01s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondById/Case_invalid_bond_id (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondById/Case_create_and_get_bond_by_id (5.01s)
    --- PASS: TestIntegrationTestSuite/TestGetQueryBondLists (4.05s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondLists/Case_create_and_get_bond_lists (4.05s)
    --- PASS: TestIntegrationTestSuite/TestGetQueryBondListsByOwner (4.05s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondListsByOwner/Case_invalid_owner_address (0.00s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondListsByOwner/Case_get_bond_lists_by_owner_address (4.05s)
    --- PASS: TestIntegrationTestSuite/TestGetQueryBondModuleBalance (4.05s)
        --- PASS: TestIntegrationTestSuite/TestGetQueryBondModuleBalance/Case_get_bond_module_balance (4.05s)
    --- PASS: TestIntegrationTestSuite/TestTxCancelBond (2.01s)
        --- PASS: TestIntegrationTestSuite/TestTxCancelBond/Case_without_bond_id (0.00s)
        --- PASS: TestIntegrationTestSuite/TestTxCancelBond/Case_cancel_bond (2.01s)
    --- PASS: TestIntegrationTestSuite/TestTxCreateBond (2.03s)
        --- PASS: TestIntegrationTestSuite/TestTxCreateBond/Case_without_deposit (0.00s)
        --- PASS: TestIntegrationTestSuite/TestTxCreateBond/Case_create_bond (2.02s)
    --- PASS: TestIntegrationTestSuite/TestTxRefillBond (2.02s)
        --- PASS: TestIntegrationTestSuite/TestTxRefillBond/Case_without_refill_amount_and_bond_id (0.00s)
        --- PASS: TestIntegrationTestSuite/TestTxRefillBond/Case_refill_bond (2.02s)
    --- PASS: TestIntegrationTestSuite/TestTxWithdrawAmountFromBond (2.03s)
        --- PASS: TestIntegrationTestSuite/TestTxWithdrawAmountFromBond/Case_without_withdraw_amount_and_bond_id (0.00s)
        --- PASS: TestIntegrationTestSuite/TestTxWithdrawAmountFromBond/Case_withdraw_amount_from_bond (2.03s)
PASS
ok      github.com/tharsis/ethermint/x/bond/client/testutil     37.633s
```

```bash
 go test -v github.com/tharsis/ethermint/x/bond/keeper
=== RUN   TestParams
--- PASS: TestParams (0.00s)
=== RUN   TestKeeperTestSuite
=== RUN   TestKeeperTestSuite/TestGrpcGetBondsByOwner
=== RUN   TestKeeperTestSuite/TestGrpcGetBondsByOwner/Case_empty_request_
=== RUN   TestKeeperTestSuite/TestGrpcGetBondsByOwner/Case_Get_Bond_By_Owner_
=== RUN   TestKeeperTestSuite/TestGrpcGetModuleBalance
=== RUN   TestKeeperTestSuite/TestGrpcGetModuleBalance/Case_empty_request_
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondBondId
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondBondId/Case_empty_request_
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondBondId/Case_Get_Bond_By_ID_
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondsList
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondsList/Case_empty_request_
=== RUN   TestKeeperTestSuite/TestGrpcQueryBondsList/Case_Get_Bonds_
=== RUN   TestKeeperTestSuite/TestGrpcQueryParams
=== RUN   TestKeeperTestSuite/TestGrpcQueryParams/Case_Get_Params_
--- PASS: TestKeeperTestSuite (0.02s)
    --- PASS: TestKeeperTestSuite/TestGrpcGetBondsByOwner (0.01s)
        --- PASS: TestKeeperTestSuite/TestGrpcGetBondsByOwner/Case_empty_request_ (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcGetBondsByOwner/Case_Get_Bond_By_Owner_ (0.00s)
    --- PASS: TestKeeperTestSuite/TestGrpcGetModuleBalance (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcGetModuleBalance/Case_empty_request_ (0.00s)
    --- PASS: TestKeeperTestSuite/TestGrpcQueryBondBondId (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcQueryBondBondId/Case_empty_request_ (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcQueryBondBondId/Case_Get_Bond_By_ID_ (0.00s)
    --- PASS: TestKeeperTestSuite/TestGrpcQueryBondsList (0.01s)
        --- PASS: TestKeeperTestSuite/TestGrpcQueryBondsList/Case_empty_request_ (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcQueryBondsList/Case_Get_Bonds_ (0.00s)
    --- PASS: TestKeeperTestSuite/TestGrpcQueryParams (0.00s)
        --- PASS: TestKeeperTestSuite/TestGrpcQueryParams/Case_Get_Params_ (0.00s)
PASS
ok      github.com/tharsis/ethermint/x/bond/keeper      0.070s
```


```bash
$ go test -v github.com/tharsis/ethermint/x/bond       
=== RUN   TestItCreatesModuleAccountOnInitBlock
--- PASS: TestItCreatesModuleAccountOnInitBlock (0.01s)
PASS
ok      github.com/tharsis/ethermint/x/bond     0.046s
```

**Comment by gsk967** [comment]
@anilCSE  pr changes are done , can you review it again

**Review by anilCSE — APPROVED** [review]
Lgtm

**Comment by gsk967** [comment]
@ashwinphatak  can you please review it

**Review by ashwinphatak — APPROVED** [review]

**Pull request merged**